### PR TITLE
fix search functionality in features list

### PR DESCRIPTION
**Pull request by imatiach-msft**
fixes 1.) search functionality not updating list of features (not working) 2.) random synchronization issues by calling updateSelection after setState completes 3.) when changing selected features that are searched for, previously selected features should not get wiped out

### CODE REVIEW

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx [hunk 1]
```diff
@@ -123,8 +130,14 @@ export class FeatureList extends React.Component<
     this.updateSelection();
   }
 
-  public componentDidUpdate(prevProps: IFeatureListProps): void {
-    if (this.props.importances !== prevProps.importances) {
+  public componentDidUpdate(
+    prevProps: IFeatureListProps,
+    prevState: IFeatureListState
```
Comment (xuke444) [inline libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx:135]: we should not monitor state in component did update, this is circular refreence.
Can we do action directly when we set state?

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx [hunk 2]
```diff
@@ -107,7 +107,14 @@ export class FeatureList extends React.Component<
...
     this._selection = new Selection({
       onSelectionChanged: (): void => {
-        const newSelectedFeatures = this.getSelectionDetails();
+        let newSelectedFeatures = this.getSelectionDetails();
+        const oldSelectedFeaturesNotSearched = this.state.selectedFeatures.filter(
+          (oldSelectedFeature) =>
+            !this.state.searchedFeatures.includes(oldSelectedFeature)
+        );
+        newSelectedFeatures = newSelectedFeatures.concat(
+          oldSelectedFeaturesNotSearched
```
Comment (xuke444) [inline libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx:116]: why do we need to concat non-searched feature?

**Review by xuke444 — APPROVED** [review]

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx (hunk 2 above)
Comment (imatiach-msft) [reply libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx:116]: so the selection is only for the active/searched items in the features list.  If there were items already selected before search, and this logic is removed, suddenly they get unselected when a user changes selection for a searched feature.

File: libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx (hunk 1 above)
Comment (imatiach-msft) [reply libs/error-analysis/src/lib/ErrorAnalysisDashboard/Controls/FeatureList/FeatureList.tsx]: done!

**Review by xuke444 — APPROVED** [review]

**Pull request merged**